### ISSUE TITLE
Tests on non-yaml output (stdout=None) are false positives

### ISSUE THREAD

**Issue by rendaw** [kislyuk/yq#12]
They all test against `""` and pass, but that's because stdout is not actually being captured in the BytesIO/StringIO - this test fails:
```
self.assertEqual(self.run_yq("foo", ["."]), "\"foo\"")
```

**Comment by kislyuk** [comment]
P.S. The tests are not false positives. They test for the fact that the subprocess does not crash.

**Comment by rendaw** [comment]
Thanks for the work on this project.

I took the fact that you were asserting the return value was `""` to mean that the `""` was significant - if it's just testing against crashes it might be more indicative to remove the `assertEqual`.

**Comment by kislyuk** [comment]
Fair point, thanks. I'll try to make it clearer next time I touch the tests.

Feel free to open PRs to improve the test suite. I usually look to keep things as simple and clear as possible, but I try to be flexible with contributions.
